src/services/LibraryService.ts

```diff
@@ -89,4 +102,33 @@ export class LibraryService implements Service<fhir4.Library> {
 
     return createLibraryPackageBundle(library[0], req.query['include-terminology'] ?? false);
   }
+
+  /**
+   * result of sending a POST request to:
+   * {BASE_URL}/4_0_1/Library/$submit or {BASE_URL}/4_0_1/Library/:id/$submit
+   * POSTs a new artifact in "draft" status. The operation results in an error if the artifact
+   * does not have status set to "draft."
+   */
+  async submit(args: RequestArgs, { req }: RequestCtx) {
+    logger.info(`${req.method} ${req.path}`);
+
+    const contentType: string | undefined = req.headers['content-type'];
+    checkContentTypeHeader(contentType);
+
+    const resource = req.body;
+    checkExpectedResourceType(resource.resourceType, 'Library');
+
+    // check for "draft" status on the resource
+    if (resource.status !== 'draft') {
+      throw new BadRequestError(`The artifact must be in 'draft' status.`);
+    }
+
+    const res = req.res;
+    // create new resource with server-defined id
+    resource['id'] = uuidv4();
+    await createResource(resource, 'Library');
+    res.status(201);
+    const location = `${constants.VERSIONS['4_0_1']}/Library/${resource.id}`;
+    res.set('Content-Location', `${process.env.HOST}:${process.env.PORT}/${location}`);
```

Comment:
for this `content-location` header, I'm guessing we want either the whole url `http://localhost:3000/4_0_1/Library/:id` or just the endpoint `/4_0_1/Library/:id`. I'd lean towards the latter since that's what we do in deqm-test-server. Also unclear if this should be `Content-Location` or just `Location` like it is in the test server for simple resource creates. Ugh... so many unknowns with this stuff.

Reply:
It also looks like the headers get lowercased when they go back through express. Does it make sense to change this to just `content-location`?

Reply:
This snippet is what happens in the `node-fhir-server-core` `create`:
<img width="708" alt="Screen Shot 2023-02-13 at 10 35 12 AM" src="https://user-images.githubusercontent.com/87094479/218501735-c2a00e77-2276-4812-9519-78a5f7ddacc2.png">

Looks like both `Content-Location` and `Location` get set, so maybe we should set both?

Reply:
Yeah I think both seems like the safest option :)

Reply:
deqm-test-server only uses `Location`. The Content-Location appears to only be for version-specific reads which we don't support. Location in deqm-test-server appears to only be the slug, not the full url

Reply:
Actually, idk where they even got `Content-Location` from. See FHIR docs, says to use `Location` for both: https://build.fhir.org/http.html#create